### PR TITLE
Docs: Consolidate guides for build customization.

### PR DESCRIPTION
**Pull request by gorzelinski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Docs: Consolidate guides for build customization. Closes https://github.com/cksource/ckeditor5-internal/issues/1868.

---

### Additional information

The other PR should be merged after this one: https://github.com/cksource/collaboration-features/pull/5350.

### CODE REVIEW

**Comment by gorzelinski** [comment]
I must add some remarks here.

Adding any plugin to the build from the online builder causes a TypeScript error:
`Class static side 'typeof Editor' incorrectly extends base class static side 'typeof ClassicEditor'.`

There is a file with custom declarations in the build folder - `ckeditor.d.ts.`

```ts
```

The array with built-in plugins has a specific set of plugins.

**Comment by gorzelinski** [comment]
I must add some remarks here.

Adding any plugin to the build from the online builder (like in the [guide](http://127.0.0.1:8080/ckeditor5/38.1.1/installation/getting-started/quick-start-other.html)) causes a TypeScript error:

`Class static side 'typeof Editor' incorrectly extends base class static side 'typeof ClassicEditor'.`

There is a file with custom declarations in the `build` folder - `ckeditor.d.ts`

```ts
declare class Editor extends ClassicEditor {
    static builtinPlugins: (typeof TextTransformation | typeof Autoformat | typeof Bold | typeof Italic | typeof BlockQuote | typeof CloudServices | typeof Essentials | typeof Paragraph | typeof Heading | typeof Image | typeof ImageCaption | typeof ImageStyle | typeof ImageToolbar | typeof ImageUpload | typeof Indent | typeof Link | typeof List | typeof MediaEmbed | typeof PasteFromOffice | typeof Table | typeof TableToolbar)[];
    static defaultConfig: {
        toolbar: {
            items: string[];
        };
        language: string;
        image: {
            toolbar: string[];
        };
        table: {
            contentToolbar: string[];
        };
    };
}
```

The array with built-in plugins seems to have a specific set of plugins. Trying to add anything seems to break the union type there. You can change the `.d.ts` file, but you'll get the same error. I'll be debugging this problem.

File: docs/installation/plugins/installing-plugins.md [hunk 1]
```diff
@@ -32,249 +28,220 @@ In order to start developing CKEditor 5 you will require:
...
+			]
+		}
+	};
+}
+
+export default Editor;
+```
 
-import { Alignment } from '@ckeditor/ckeditor5-alignment'; // <--- ADDED
+## Adding a JavaScript plugin
```
Comment (apadol-cksource) [inline docs/installation/plugins/installing-plugins.md:167]: In most cases, developers will use plugins provided by CKEditor and then the section will cause confusion. Let's move it to the end of the article.

File: docs/installation/plugins/installing-plugins.md [hunk 2]
```diff
@@ -32,249 +28,220 @@ In order to start developing CKEditor 5 you will require:
...
 
-export default class ClassicEditor extends ClassicEditorBase {}
+The CKEditor is a TypeScript project, and all plugins also use TypeScript. However, there are ways to use JavaScript packages with the editor.
```
Comment (apadol-cksource) [inline docs/installation/plugins/installing-plugins.md:169]:
```suggestion
The CKEditor 5 is a TypeScript project, and all plugins provided by CKEditor 5 also use TypeScript. However, there are ways to use JavaScript packages with the editor.
```

**Review by apadol-cksource — CHANGES_REQUESTED** [review]

File: docs/installation/plugins/installing-plugins.md [hunk 3]
```diff
@@ -32,249 +28,220 @@ In order to start developing CKEditor 5 you will require:
 	**NOTE:** The above rule rule does not apply to packages named `@ckeditor/ckeditor5-dev-*`.
 </info-box>
 
-## Adding a plugin to a build
+## Creating an editor base
```
[the remark this answers is not in the document]
Comment (gorzelinski) [reply docs/installation/plugins/installing-plugins.md:31]: I wasn't sure about this subheading either. Tbh, it was more of a placeholder. I'll change it.

File: docs/installation/plugins/installing-plugins.md [hunk 4]
```diff
@@ -32,249 +28,220 @@ In order to start developing CKEditor 5 you will require:
...
+	You can also give [Yarn](https://yarnpkg.com/lang/en/) a try.
+</info-box>
+
+## Adding a plugin to a configuration
+
+To add a plugin to your editor, you need to follow three steps:
+
+1. Import the installed package in the file with the CKEditor configuration.
+2. Add the imported plugin to the list of plugins. There are two ways to achieve that: using the {@link module:core/editor/editor~Editor.builtinPlugins `builtinPlugins`} property or passing a plugin to the {@link module:core/editor/editor~Editor.create `create()`} method. Adding a plugin through the property lets you automatically enable it in all editor instances using this editor class. Passing the plugin to the method will affect only one instance.
+3. Display proper UI for the installed plugin to let users use it.
```
[the remark this answers is not in the document]
Comment (gorzelinski) [reply docs/installation/plugins/installing-plugins.md:63]: I forgot about plugins that don't have a UI - good point.

**Review by apadol-cksource — APPROVED** [review]

**Pull request merged**